lucet-runtime/lucet-runtime-internals/src/future.rs

```diff
@@ -49,65 +54,67 @@ impl Vmctx {
     ///
     /// Note that this method may only be used if `Instance::run_async` was used to run the VM,
     /// otherwise it will terminate the instance with `TerminationDetails::BlockOnNeedsAsync`.
-    pub fn block_on<'a, R>(&'a self, f: impl Future<Output = R> + Send + 'a) -> R
+    pub fn block_on<R>(&self, f: impl Future<Output = R> + Send) -> R
```

Comment:
Now that block on is zero-overhead when polling is unnecessary, I'd imagine that it would be possible to make `async fn` hostcalls possible by changing the macro. I'll look into this later on today.

Reply:
Nope, this wouldn't work, because it would be a sharp edge depending on if the future is within the guest execution context